src/support/check_header.asm

```diff
@@ -398,11 +402,53 @@ choose_region:
   log_string("Finished region:")
   hex.b r10
   ret
+  
+check_bsx:
+  log_string("Checking BSX")
+  // buf[addr - 14] == 'Z' && buf[addr - 11] == 'J' &&
+  check_value_equality(rambuf, 0x5A) // Check if 'Z'
+  jp nz, not_bsx
+  check_value_equality(rambuf + 3, 0x4A) // Check if 'J'
+  jp nz, not_bsx
+
+  // ((buf[addr - 13] >= 'A' && buf[addr - 13] <= 'Z') || (buf[addr - 13] >= '0' && buf[addr - 13] <= '9')) &&
+  // Condition one
+  check_value_equality(rambuf + 1, 0x41) // 'A'
+  jp c, condition_two
+  check_value_equality(rambuf + 1, 0x5A) // 'Z'
+  jp c, condition_three // Carry or Zero means <=
+  jp z, condition_three
+
+  condition_two:
+  check_value_equality(rambuf + 1, 0x30) // '0'
+  jp c, not_bsx
+  check_value_equality(rambuf + 1, 0x39) // '9'
+  jp c, condition_three // Carry or Zero means <=
+  jp z, condition_three
+
+  // (buf[addr + Company] == 0x33 || (buf[addr - 10] == 0x00 && buf[addr - 4] == 0x00))
+  condition_three:
+  check_value_equality(dev_id_addr, 0x33)
+  jp z, is_bsx
+
+  check_value_equality(rambuf + 4, 0)
+  jp nz, not_bsx
+  check_value_equality(rambuf + 10, 0)
+  jp nz, not_bsx
+
+  is_bsx:
+  // Set chip_type lower nibble to 3
+  or r12,#0x30
```

Comment:
That is not the lower nibble

Reply:
This doesn't appear to match the MiSTer logic. If the above logic passes, then the ROM type is denoted as 3, which is the low nibble of the chip type value.

The upper nibble is only set if the file is detected to be a BSX BIOS, which is not checked for here.